lib/rbac/src/jwt.rs

```diff
@@ -25,18 +25,44 @@ pub struct Claims {
 
 pub type PayloadClaim = HashMap<JsonPath, ValueVariants>;
 
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct KeyValuePair {
+    key: JsonPath,
+    value: ValueVariants,
+}
+
+impl KeyValuePair {
+    pub fn to_condition(&self) -> Condition {
+        Condition::Field(FieldCondition::new_match(
+            self.key.clone(),
+            Match::new_value(self.value.clone()),
+        ))
+    }
+}
+
 #[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
 pub struct ValueExists {
-    pub collection: String,
-    pub key: JsonPath,
-    pub value: ValueVariants,
+    collection: String,
+    matches: Vec<KeyValuePair>,
```

Comment:
Maybe use `HashMap<JsonPath, ValueVariants>` to match with the `payload` claim?

Reply:
Not the same. We might want to check multiple values of the same field. Cause field can be array of values